### PR TITLE
LP-2303 LP-2294 Unit test and linting for assessment and autoscore ORA command

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
### Description

**[LP-2303](https://philanthropyu.atlassian.net/browse/LP-2303)** Autoscore ORA command (Unit test only)
**[LP-2294](https://philanthropyu.atlassian.net/browse/LP-2294)** Fix all linting errors for assessment

**Notes**

- [x] Create a fixture to mute all model signals in edx-platform/conftest.py so that any pytest can use it, when required.
- [ ] Change in `conftest.py` is a core change, I will log it once PR is approved
- [x] All unit test for autoscore ORA command and openedx assessment included
- [x] Linitng of autoscore ORA command and openedx assessment included
- [x] iSort ran on all files
- [x] Improvement in `common/djangoapps/student/tests/factories.py` is a core change, which was initially made by nofal, so no need to log it.
- [x] Fixed following linting issue particular to LP-2294

```
************* Module edx-platform.openedx.features.assessment.tests.factories
openedx/features/assessment/tests/factories.py:17: [C1001(old-style-class), StudentItemFactory.Meta] Old-style class defined.
openedx/features/assessment/tests/factories.py:31: [C1001(old-style-class), SubmissionFactory.Meta] Old-style class defined.
************* Module edx-platform.openedx.features.assessment.tests.test_helpers
openedx/features/assessment/tests/test_helpers.py:136: [W0212(protected-access), AssessmentHelperModuleStoreTestCase.test_get_submissions_to_autoscore_by_enrollment_successfully] Access to a protected member _get_submissions_to_autoscore_by_enrollment of a client class
openedx/features/assessment/tests/test_helpers.py:154: [W0212(protected-access), AssessmentHelperModuleStoreTestCase.test_get_submissions_to_autoscore_by_enrollment_no_submissions] Access to a protected member _get_submissions_to_autoscore_by_enrollment of a client class
openedx/features/assessment/tests/test_helpers.py:174: [W0212(protected-access), AssessmentHelperModuleStoreTestCase.test_get_submissions_to_autoscore_by_enrollment_course_without_ora] Access to a protected member _get_submissions_to_autoscore_by_enrollment of a client class
openedx/features/assessment/tests/test_helpers.py:223: [W0212(protected-access), AssessmentHelperTestCases.test_log_multiple_submissions_info_with_multiple_submissions] Access to a protected member _log_multiple_submissions_info of a client class
openedx/features/assessment/tests/test_helpers.py:233: [W0212(protected-access), AssessmentHelperTestCases.test_log_multiple_submissions_info_with_no_submissions] Access to a protected member _log_multiple_submissions_info of a client class
openedx/features/assessment/tests/test_helpers.py:16: [W0611(unused-import), ] Unused AssessmentPart imported from openassessment.assessment.models
openedx/features/assessment/tests/test_helpers.py:21: [W0611(unused-import), ] Unused StudentItemFactory imported from openedx.features.assessment.tests.factories

-----------------------------------
Your code has been rated at 9.62/10
```

### CODE REVIEW

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: openedx/features/assessment/tests/test_helpers.py [hunk 1]
```diff
@@ -1,174 +1,296 @@
...
-        self.block_usage_locator = BlockUsageLocator.from_string(unicode(self.all_ora_in_course[0].location))
-
-    @mock.patch('openedx.features.assessment.helpers.log.info')
-    def test_log_multiple_submissions_info_with_multiple_submissions(self, mock_log_info):
-        submissions = SubmissionFactory.create_batch(2)
-        submission_ids = map(str, [submission.id for submission in submissions])
+    def _create_enrollment_submission(self, submission_date, course_id, canceled_one_submission=False,
+                                      exclude_submissions=False):
+        """
+        A helper function for tests, which create course enrollment, and perform submission for ORAs in course.
```
Comment (harris-irfan) [inline openedx/features/assessment/tests/test_helpers.py:63]: *creates
*performs

File: openedx/features/philu_utils/constants.py [hunk 2]
```diff
@@ -1,2 +1,11 @@
...
+"""
+All constants for Philu utils
+"""
+AWS_S3_PATH = 'https://s3.amazonaws.com/{bucket}/{key}'  # S3 complete path to file
+
+COURSE_CHILD_STRUCTURE = {
+    "course": "chapter",
+    "chapter": "sequential",
+    "sequential": "vertical",
+    "vertical": "html",
```
Comment (harris-irfan) [inline openedx/features/philu_utils/constants.py:10]: single qoutes?

**Review by harris-irfan — CHANGES_REQUESTED** [review]

**Comment by harris-irfan** [comment]
Aren't we supposed to either suppress the warnings in the pylint report, or solve them?

File: common/djangoapps/philu_commands/management/commands/tests/test_autoscore_ondemand_course_ora.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        course_id = enrollment.course_id.to_deprecated_string()
+        submission = SubmissionFactory(
+            student_item__student_id=AnonymousUserIdFactory(course_id=course_id, user=enrollment.user),
+            student_item__course_id=enrollment.course_id.to_deprecated_string(),
+            student_item__item_id=enrollment.user.username
+        )
+        workflow_api.create_workflow(submission.uuid, ["training", "peer", "self"])
+
+        if enrollment.course_overview.self_paced:
+            expected_submission_uuids.insert(0, unicode(submission.uuid))
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply common/djangoapps/philu_commands/management/commands/tests/test_autoscore_ondemand_course_ora.py]: I have created a new test for testing these two states

**Review by mabdullahz — CHANGES_REQUESTED** [review]

File: common/djangoapps/student/tests/factories.py [hunk 4]
```diff
@@ -215,7 +215,12 @@ class Meta(object):
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply common/djangoapps/student/tests/factories.py:215]: This is an already written code by us, I have just updated it, docstring is added for our own convenience, not for the sake of linting. Moving forward we can create factories in our code, but where will we keep factories?

File: openedx/features/philu_utils/tests/mixins.py [hunk 5]
```diff
@@ -13,3 +23,42 @@ def setUpClass(cls):
     def tearDownClass(cls):
         clear_philu_theme()
         super(PhiluThemeMixin, cls).tearDownClass()
+
+
+class CourseAssessmentMixin(object):
+    """
+    Base open assessment testcase class, with common methods to create courses dynamic, including ORA.
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply openedx/features/philu_utils/tests/mixins.py:30]: Updated to `to create course dynamically.`

File: common/djangoapps/philu_commands/management/commands/tests/test_autoscore_ondemand_course_ora.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
+"""
+Test for autoscore ondemand course ora command.
+"""
+import mock
+import pytest
+from django.core.management import call_command
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply common/djangoapps/philu_commands/management/commands/tests/test_autoscore_ondemand_course_ora.py:6]: `iSort` wan ran on all files

File: common/djangoapps/philu_commands/management/commands/tests/test_autoscore_ondemand_course_ora.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+        instructor_paced_course = CourseOverviewFactory(self_paced=False)
+        enrollments = CourseEnrollmentFactory.create_batch(5, course_id=instructor_paced_course.id)
+        return enrollments
+
+
+@pytest.mark.django_db
+@mock.patch('philu_commands.management.commands.autoscore_ondemand_course_ora.find_and_autoscore_submissions')
+@mock.patch('philu_commands.management.commands.autoscore_ondemand_course_ora.log.info')
+# pylint: disable=redefined-outer-name
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply common/djangoapps/philu_commands/management/commands/tests/test_autoscore_ondemand_course_ora.py:45]: We cannot, this is how we use pytest fixture (another way is to use decorator) and it should be named exactly how it was defined.

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: common/djangoapps/student/tests/factories.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mabdullahz) [reply common/djangoapps/student/tests/factories.py:215]: Makes sense. I think we can look for a better place moving forward :)

**Review by mabdullahz — APPROVED** [review]

**Review by harris-irfan — APPROVED** [review]

**Pull request merged**